base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs

```diff
@@ -547,14 +696,79 @@ impl OutputSql {
         diesel::insert_into(outputs::table).values(self.clone()).execute(conn)?;
         Ok(())
     }
+}
+
+impl Encryptable<Aes256Gcm> for NewOutputSql {
+    fn encrypt(&mut self, cipher: &Aes256Gcm) -> Result<(), AeadError> {
+        self.spending_key = encrypt_bytes_integral_nonce(&cipher, self.spending_key.clone())?;
+        Ok(())
+    }
+
+    fn decrypt(&mut self, cipher: &Aes256Gcm) -> Result<(), AeadError> {
+        self.spending_key = decrypt_bytes_integral_nonce(&cipher, self.spending_key.clone())?;
+        Ok(())
+    }
+}
 
-    /// Return all unencumbered outputs
-    #[cfg(test)]
+#[derive(Clone, Debug, Queryable, Identifiable, PartialEq)]
+#[table_name = "outputs"]
+struct OutputSql {
+    id: i32,
+    commitment: Option<Vec<u8>>,
+    spending_key: Vec<u8>,
+    value: i64,
+    flags: i32,
+    maturity: i64,
+    status: i32,
+    tx_id: Option<i64>,
+    hash: Option<Vec<u8>>,
+}
+
+impl OutputSql {
+    // This function is to update the values of hte hash field if its missing, It will check if the hash is missing, add
+    // this and update the field. ToDo remove this post testnet 1
+    fn migrate(conn: &SqliteConnection, cipher: Option<Aes256Gcm>) -> Result<(), OutputManagerStorageError> {
+        // Update Hashes
+        let mut ou_array: Vec<OutputSql> = outputs::table.filter(outputs::hash.is_null()).load(conn)?;
+        if let Some(cipher_inner) = cipher.clone() {
+            for o in ou_array.iter_mut() {
+                o.decrypt(&cipher_inner)
+                    .map_err(|_| OutputManagerStorageError::AeadError("Decryption Error".to_string()))?;
+            }
+        }
+        for output in ou_array {
+            // This should only happen on database migration as the hash will then be empty
+            // TODO remove this as this is temp migration code
+            let ou = DbUnblindedOutput::try_from(output.clone())?;
+            diesel::update(outputs::table.filter(outputs::id.eq(&output.id)))
+                .set(outputs::hash.eq(Some(ou.hash.clone())))
+                .execute(conn)?;
```

Comment:
Is there a transaction around this operation in the calling method?

Reply:
There isn't I have an open issue about adding transactions to the wallet database that I have not gotten to: https://github.com/tari-project/tari/issues/908